tests/mir_test_framework/canonical_window_manager.cpp

```diff
@@ -418,21 +421,30 @@ void msh::CanonicalWindowManagerPolicy::handle_delete_surface(std::shared_ptr<ms
     fullscreen_surfaces.erase(surface);
     bool const is_active_surface{surface.lock() == active_surface()};
 
-    auto& info = tools->info_for(surface);
-
-    if (auto const parent = info.parent.lock())
+    try
     {
-        auto& siblings = tools->info_for(parent).children;
+        auto& info = tools->info_for(surface);
 
-        for (auto i = begin(siblings); i != end(siblings); ++i)
+        if (auto const parent = info.parent.lock())
         {
-            if (surface.lock() == i->lock())
+            auto& siblings = tools->info_for(parent).children;
+
+            for (auto i = begin(siblings); i != end(siblings); ++i)
             {
-                siblings.erase(i);
-                break;
+                if (surface.lock() == i->lock())
+                {
+                    siblings.erase(i);
+                    break;
+                }
             }
         }
     }
+    catch (std::out_of_range e)
```

Comment:
You should generally `catch` by const-reference, rather than by value.

Reply:
@wmww you've responded a couple of times to the "needs info" comment, but not reacted to the "needs fixing". Have you overlooked this?

Reply:
Was too busy to check out another branch and fix at the time, then forgot. Just pushed a fix now.